libs/chain-signatures/contract/src/lib.rs

```diff
@@ -810,206 +579,133 @@ impl VersionedMpcContract {
     #[init(ignore_state)]
     #[handle_result]
     pub fn migrate() -> Result<Self, Error> {
-        let old: VersionedMpcContract =
-            env::state_read().ok_or(InvalidState::ContractStateIsMissing)?;
-        match old {
-            VersionedMpcContract::V0(mpc_contract_v0) => {
-                Ok(VersionedMpcContract::V1(MpcContractV1 {
-                    config: ConfigV1::default(),
-                    protocol_state: mpc_contract_v0.protocol_state,
-                    pending_requests: mpc_contract_v0.pending_requests,
-                    request_by_block_height: Vector::new(StorageKey::RequestsByTimestamp),
-                    proposed_updates: ProposedUpdates::default(),
-                }))
+        if let Some(legacy_contract::VersionedMpcContract::V1(state)) =
+            env::state_read::<legacy_contract::VersionedMpcContract>()
+        {
+            // migrate config
+            let mut config = Config::default();
+            config.request_timeout_blocks = state.config.request_timeout_blocks;
+            config.max_num_requests_to_remove = state.config.max_num_requests_to_remove;
+            // migrate signature requests:
+            let mut request_by_block_height: Vector<(BlockHeight, SignatureRequest)> =
+                Vector::new(StorageKey::RequestsByTimestamp);
+            let mut pending_requests: LookupMap<SignatureRequest, YieldIndex> =
+                LookupMap::new(StorageKey::PendingRequests);
+            for (created, request) in &state.request_by_block_height {
```

Comment:
Could this run out of gas?

Reply:
Great question, I don't know.

Reply:
deferred to https://github.com/Near-One/mpc/issues/241